### ISSUE TITLE
Sync Edit and Preview Scrollbars when editing a blogpost

### ISSUE THREAD

**Issue by trickstival** [forem/forem#10380]
**Is your feature request related to a problem? Please describe.**

Yes. It is a little bit annoying when I'm writing a blogpost and I want to see my markdown rendered.
I always have to scroll down to the place I was editing every time I switch from the `Edit` to the `Preview` tab.

**Describe the solution you'd like**

It would be really cool if I could split the page between both of them, so I can see my changes applied in real time.
Another solution would be to sync the scrollbar y position between both tabs, but it would be hard to find out the y value for the `Preview` tab, since it's rendered content.

**Describe alternatives you've considered**

Maybe a quick alternative solution would be to write the markdown in another editor and paste the text on the dev.to editor when I want to publish, but it would be better to have it builtin.

**Comment by twinsfan421** [comment]
I can work on this

**Comment by fdoxyz** [comment]
Thanks @twinsfan421. Are you looking into making the sync of the scrollbars work? I think this is a great enhancement that would greatly benefit the publishing experience. Do you have an idea in mind to how to make this work?

The split screen option for the editor is something we haven't had a chance to work on from the design perspective though. As the team will be a little [slower to respond this coming week](https://dev.to/devteam/the-dev-forem-team-is-taking-a-mental-health-holiday-1ldl) it maybe best to wait if that's what you rather do.

**Comment by twinsfan421** [comment]
Hello @fdoxyz.  I am open to syncing the scrollbars or the split screen option.  No worries on the slower response next week.  I can wait to hear back on what option/design you would prefer before I take a deeper look. Thanks

**Comment by fdoxyz** [comment]
Until we have a design/product decision in place for a split view I think the alternative solution (which is actually the title of the issue) is a great option as it would be a very useful enhancement of the current experience. This would mean syncing the scroll in both tabs.

As originally mentioned, it is tricky to implement because the Preview tab has rendered content. The current offset of the Edit tab will most likely not match the Preview tab (i.e. adding 2 tall images in markdown represent one line each in markdown but can take more than twice the viewport height).

A possible solution would be to keep track the first portion of visible content in the Edit tab `textarea` (`id="article_body_markdown"`) and then make sure the scroll in the Preview tab has that element visible (ideally both ways as well after scrolling in the Preview tab). The difficulties I see in order to implement this are the following:
- I (personally) don't know if it's possible to know what's the first portion of visible text in a `textarea` element using JS or a query selector
- Being able to scroll the Preview tab when the first portion of visible is
  - Plain text (`Sample text in a doc`)
  - Markdwon tag (`## Header text` or worse a multi-line codeblock)
  - An image annotation (`![IMG_1276](https://de...1oimrzh.PNG)`)
  - Custom liquid tag (`{% twitter 834439977220112384 %}`)

I'm swapping the `ready for dev` label for `discussion` as it would probably be good to open for discussion the possible ways to make this work. All ideas/suggestions are welcome 🙂

**Comment by ludwiczakpawel** [comment]
Split screen will be definitely huuuuuge project - not sure if we're willing/ready to work through it right now, even from design perspective... And i don't necessarily think it's an ideal solution because that kinda is "power-user" thingy - I don't expect my mothers and fathers of this world to understand why the heck they can see two panels with their post.. And I'm pointing this out because we're now more thinking more holistically and less DEV (and Developers too) oriented because of [Forem](https://dev.to/devteam/for-empowering-community-2k6h). So if we ever want to have it, it should be _an option_ - so something you can trigger manually if you need to.

Syncing scrolls is IMO good idea if it's executed well - all my thoughts and concerns overlap with what already has been mentioned above so I'm not gonna repeat it.

---

Alternative solution to that problem, and that's waaaaay beyond this issue, is thinking about _WHY_ do you need to sync scrollbars in the first place? I guess it's because you check the Preview quite often and it's just easier and more smooth workflow when scrollbars are synced... But then maybe we should solve that problem instead? Here's an example of pretty outdated library for markdown text editor (I'm sure there are better alternatives, it's just the one I found):  https://simplemde.com/ - it's a hybrid of Markdown and WYSIWYG - something that could potentially be our perfect solution - no need to have "Edit | Preview" tabs because your Markdown renders as its gonna render on the page anyway (well, almost)...

**Comment by trickstival** [comment]
@ludwiczakpawel 
I really like the `WYSIWYG` experience from the editors you found.
It would be really cool to have this kind of experience while writing blogposts, but I still think syncing the scrollbars is valuable.

[stackedit](https://stackedit.io/) seems to have the same offsets on both scrollbars, since the content height is the same. Even images get rendered in the editor, which is pretty cool.

**Comment by ludwiczakpawel** [comment]
@trickstival yea, if only implementing that was trivial :)

Stackedit indeed syncs the scrollbar BUT they also render images and everything in the editor as well, not only in the preview mode. Imagine situation when you have entered a markdown for some image.. In our editor it would be one line of text (code) and in the preview it would be an actual image with some height defined. How do you sync scrollbars then? The same for our [Liquid Tags](https://docs.forem.com/frontend/liquid-tags/). But if you can find a way to make it work, then go for it :)

But yea, the ideal solution would be implementing the WYSIWYG like experience :)

**Comment by cmgorton** [comment]
Hello! Thanks for your contributions here. We have a new process moving forward to address feature requests.
TLDR: we have new internal RFC process for feature requests. Any features that are more substantial and require more thought and detail will need to be discussed in [forem.dev](forem.dev).

Given that this feature request appears to need more discussion and buy in I am going to close it here in favor of a forem.dev post. 

You can read more about our Internal RFC Process and forem.dev Discussions in [this post](https://forem.dev/foremteam/internal-rfc-process-and-forem-dev-discussions-3gl4).
